pkg/virt-controller/watch/workload-updater/workload-updater_test.go

```diff
@@ -550,6 +552,29 @@ var _ = Describe("Workload Updater", func() {
 			controller.Execute()
 			testutils.ExpectEvent(recorder, FailedChangeAbortionReason)
 		})
+
+		It("shouldn't cancel the migration if the migration object is still in running phase but the domain is ready on the target", func() {
+			vmi := api.NewMinimalVMI("testvm")
+			vmi.Namespace = k8sv1.NamespaceDefault
+			vmi.Status.Phase = v1.Running
+			vmi.Status.Conditions = append(vmi.Status.Conditions, v1.VirtualMachineInstanceCondition{
+				Type: v1.VirtualMachineInstanceIsMigratable, Status: k8sv1.ConditionTrue})
+			vmi.Status.Conditions = append(vmi.Status.Conditions, v1.VirtualMachineInstanceCondition{
+				Type:   v1.VirtualMachineInstanceMemoryChange,
+				Status: k8sv1.ConditionTrue,
+			})
+			vmi.Status.MigrationState = &v1.VirtualMachineInstanceMigrationState{
+				StartTimestamp:                 pointer.P(metav1.Now()),
+				Completed:                      false,
+				Failed:                         false,
+				TargetNodeDomainReadyTimestamp: pointer.P(metav1.Now()),
+			}
+			vmiSource.Add(vmi)
```

Comment:
This is async operation, we need to sync the state otherwise this is flaky.

Reply:
This has been replaced by the store in the controller, does it still require a sync?